extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java

```diff
@@ -191,4 +218,9 @@ private void addDerivedMetricsRow(Map<String, Object> derivedMetrics, List<Strin
     ));
   }
 
+  private String prefixKeyMapper(String prefix, String key)
+  {
+    return Strings.isNullOrEmpty(prefix) ? key : prefix + key;
```

Comment:
same here, let's not do the null checks every time, let's resolve the prefix in the constructor when we initialize the parser.

Reply:
Moved `NullOrEmpty` method as one time check in constructor with boolean variable. I think that can be only optimization we can have here.

Reply:
Done.